### PR TITLE
Redefined reportError function/User input restrictions

### PR DESCRIPTION
**Pull request by AbnerSD**
What we changed:
- User input restrictions: character limit and character restriction.
- Changed the reportError function to display a stylized div, that replaces the alert function it had before.

### CODE REVIEW

File: ui/rla_ui.js [hunk 1]
```diff
@@ -8,968 +8,996 @@
 //   (Either define this more carefully or just limit to "a-z0-9_-")
 
 
-(function(){
```
Comment (tom-votes) [inline ui/rla_ui.js]: @AbnerSD there's "diff noise" here making the real changes harder to find. I believe your editor is changing spaces to tabs, and using `"\r\n"` newlines instead of `"\n"`. I can add `?w=0` to the url to help with the diff a bit, but would you mind changing this so the git history is easier to read?

It's probably an editor setting (e.g. for replacing tabs), but you can also use something like this:

    sed -i 's/\r//' ui/rla_ui.js

File: ui/index.html [hunk 2]
```diff
@@ -1,39 +1,61 @@
...
-         <input type="text" id="auditNameBox" />
-         <button id="auditNameSaveButton">Save</button>
-      </div>
-      <div id="cvrFileUploadContainer" class="container" style="display:none">
-         <p>Upload a Cast Vote Record (CVR) file</p>
-         <form id="uploadCVRForm" method="post" enctype="multipart/form-data">
+
+<head>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
+    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.4.0/js/bootstrap.min.js"></script>
```
Comment (tom-votes) [inline ui/index.html]: `audit-conductor` should be able to run without an internet connection, so we locally load e.g. jquery with:

    <script src="jquery.js"></script>

And we serve it from the app (https://github.com/ElectionAuditWare/audit-conductor/blob/3de6dca06854a035072a8fab49a984cad7d52bb0/conductor.py#L989)

If Bootstrap's important for the change, you can serve it locally the same way.

File: ui/rla_ui.js [hunk 3]
```diff
@@ -8,968 +8,996 @@
...
-   alert('AJAX failure!: '+JSON.stringify(e));
-}
-
-})();
+(function () {
+
+	var debugMode = false; // true;
+
+	if (debugMode) {
+		//alert('Note: running in debug mode!');
```
Comment (tom-votes) [inline ui/rla_ui.js]: You can probably fix the spaces-to-tabs change with:

    sed -i 's/\t/   /' ui/rla_ui.js

I'd recommend changing your editor settings if it automatically replaces it, though, since it'll keep trying.

File: ui/rla_ui.js [hunk 4]
```diff
@@ -8,968 +8,996 @@
...
-
-// TODO: big red error box here:
-function reportError(e) {
-   alert('AJAX failure!: '+JSON.stringify(e));
-}
-
-})();
+(function () {
+
+	var debugMode = false; // true;
```
Comment (tom-votes) [inline ui/rla_ui.js]: The whole file's inside the `(function () {` and `})();` lines, so it's fine to leave it not indented.

File: ui/index.html [hunk 5]
```diff
@@ -1,39 +1,61 @@
...
-      <div id="auditNameContainer" class="container" style="display:none">
-         Enter a name for this audit:<br />
-         <input type="text" id="auditNameBox" />
-         <button id="auditNameSaveButton">Save</button>
-      </div>
-      <div id="cvrFileUploadContainer" class="container" style="display:none">
-         <p>Upload a Cast Vote Record (CVR) file</p>
-         <form id="uploadCVRForm" method="post" enctype="multipart/form-data">
+
+<head>
```
Comment (tom-votes) [inline ui/index.html]: (Basically the opposite of my advice for the JS file, below:)

Since the file started out with indentation before e.g. `<head>` and `<body>`, it's probably best to leave it that way to reduce the diff noise.

Comment (tom-votes) [reply ui/index.html]: (Opposite indentation advice, but both in the service of making the smallest possible diff)

File: ui/rla_ui.js [hunk 6]
```diff
@@ -8,968 +8,996 @@
...
+				label = newElem('label');
+				label.innerHTML = prettifyChoice(choice); // again, careful!
+				label.setAttribute('for', checkbox.id);
+				label.classList.add('choiceCheckboxLabel');
+
+				li = newElem('li');
+
+				li.appendChild(checkbox);
+				li.appendChild(label);
+				ul
```
Comment (tom-votes) [inline ui/rla_ui.js]: Looks good!

Comment (tom-votes) [inline ui/rla_ui.js]: To reduce jQuery usage, can be:

    modalclose.onclick = function() {

Like in e.g. https://github.com/ElectionAuditWare/audit-conductor/blob/3de6dca06854a035072a8fab49a984cad7d52bb0/ui/rla_ui.js#L335

Comment (tom-votes) [inline ui/rla_ui.js]: Can use `getById` instead of `document.getElementById`

https://github.com/ElectionAuditWare/audit-conductor/blob/3de6dca06854a035072a8fab49a984cad7d52bb0/ui/rla_ui.js#L56

Comment (tom-votes) [inline ui/rla_ui.js]: To reduce jQuery usage, can use `getById('errorBox')` instead of `$("#errorBox")`

File: ui/index.html [hunk 7]
```diff
@@ -1,39 +1,61 @@
...
+<body>
+    <div id="auditTypeContainer" class="container">
+        Choose the audit type:
+        <br />
+    </div>
+    <div id="auditNameContainer" class="container" style="display:none">
+        Enter a name for this audit:
+        <br />
+        <!-- Regular expression to restrict user input. We call on the limitKeypress function which limits the amount of characters in the input field which is defined by maxlength. -->
+        <input type="text" id="auditNameBox" onkeypress="return event.charCode === 0 || /[a-zA-Z\-\_\d\.]+/.test(String.fromCharCode(event.charCode));" maxlength="15" onkeypress="limitKeypress(event,this.value,maxlength)" / />
```
Comment (tom-votes) [inline ui/index.html]: Can this `onkeypress` and the `onkeypress` below be pulled out into functions in `ui/rla_ui.js`?

File: ui/style.css [hunk 8]
```diff
@@ -2,73 +2,191 @@
...
+
 /* todo: some common class betw this and 'ballot': */
-#seedContainer, #contestNameContainer, #contestTypeContainer, #finalResultContainer, #csvUploadContainer {
-}
+
+#seedContainer,
+#contestNameContainer,
+#contestTypeContainer,
+#finalResultContainer,
+#csvUploadContainer {}
```
Comment (tom-votes) [inline ui/style.css:56]: :+1: this is easier to read

**Comment by tom-votes** [comment]
Thanks @AbnerSD ! I've added some comments for next steps.

File: ui/rla_ui.js (hunk 1 above)
Comment (tom-votes) [reply ui/rla_ui.js]: Ah, after further research it looks like this may be another way to solve the `"\r\n"` problem:

https://git-scm.com/docs/git-config#Documentation/git-config.txt-coreautocrlf
https://help.github.com/en/articles/dealing-with-line-endings

In other words, set `core.autocrlf` to `true`, and it will use `"\r\n"` locally but then commit with `"\n"`.

File: ui/index.html (hunk 7 above)
Comment (AbnerSD) [reply ui/index.html]: I made a function in Javascript and passed it to the seed and audit name functions.
